### ISSUE TITLE
How to get shared clipboard and good performance?

### ISSUE THREAD

**Issue by LorenzoBettini** [quickemu-project/quickemu#933]
#### Expected behaviour

First of all, I'm only experimenting with Linux VMs. And the spice agent software is installed on the host.

I seem to understand that shared clipboard only works when using `--display spice`. Indeed, it does (only when using an X11, not when the host runs a Wayland session). However, the performance is really bad (3d effects are really slow). Then I tried `--display spice-app`, however, in that case, the VM window flashes and has serious graphical problems that do not even allow me to log into the VM.

Long story short: is there a way to have a shared clipboard in the Linux VM without sacrificing performance?

#### Quickemu output

Run `quickemu` or `quickget` and paste the output here.

```
❯ quickemu --version                                           
4.9.1
```

#### Linux Distribution & Kernel

Run the following and paste the output.

```bash
lsb_release --all

LSB Version:    n/a
Distributor ID: EndeavourOS
Description:    EndeavourOS Linux
Release:        rolling
Codename:       rolling

uname -a

Linux eos-optiplex 6.6.16-1-lts #1 SMP PREEMPT_DYNAMIC Mon, 05 Feb 2024 21:20:21 +0000 x86_64 GNU/Linux
```

**Comment by TuxVinyards** [comment]
You are never going to get fast slick graphical desktop performance.  The Spice feature is useful for pasting the file. It ends up in the Downloads folder.  You can use qqX to set up a shared drive that all the different VM's can access. Also works in SDL mode too ...  That's about it.

https://github.com/HikariKnight has a 3D pass through project you might want to look at.  I haven't tried it.  More aimed at gamers.

**Comment by LorenzoBettini** [comment]
> You are never going to get fast slick graphical desktop performance. 

You mean with spice, right?
With the default `--display sdl` the performance are great.

**Comment by TuxVinyards** [comment]
> "With the default --display sdl the performance are great."

Well, I wouldn't say that the performance is ever going to bowl you over, but yes, I think it is a bit crisper with SDL.  And you can set things up to launch at a given screen size too ...

**Comment by LorenzoBettini** [comment]
I wouldn't call it just a bit crispier ;)
On my computers, it doesn't even feel like a virtual machine: it's really smooth. And only with an Intel HD Graphics.
When using the spice display instead things degrade a lot.

**Comment by TuxVinyards** [comment]
Good to hear :+1:

**Comment by muhmud** [comment]
I have the same issue, `sdl` works really great, like native performance, but I can't get the clipboard to work, which I really need. Meanwhile, I can copy/paste with `spice`, but the performance is disappointing.

Is there anyway to get shared clipboard to work in `sdl`?

**Comment by TuxVinyards** [comment]
No. That's it.  Those are the tools that have. But at least we have those ....
